.github/workflows/ci.yml

```diff
@@ -607,4 +612,37 @@ jobs:
         run: cargo cache --autoclean-expensive
       - name: Stop sccache
         if: always() && steps.sccache.conclusion == 'success'
-        run: sccache --stop-server || true
\ No newline at end of file
+        run: sccache --stop-server || true
+
+  upload-coverage:
+    runs-on: [self-hosted, 8vcpu-16ram-ubuntu22-namada-x86]
+    container: 
+      image: ghcr.io/heliaxdev/namada-ci:namada-main
+    timeout-minutes: 20
+    needs: [test-unit, test-integration]
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        if: ${{ github.event_name != 'pull_request_target' }}
+      - name: Checkout PR
+        uses: actions/checkout@v4
+        if: ${{ github.event_name == 'pull_request_target' }}
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
we dont need to checkout the repo, just download the artifacts and upload

Reply:
[542992b](https://github.com/anoma/namada/pull/3704/commits/542992b5193d3a66d410c24a6f23abb0b33a3840)